kokoro/builds/build.sh

```diff
@@ -208,6 +214,32 @@ if [ -n "$1" ]; then
     popd > /dev/null
   fi
 
+  # Analyze include-what-you-use results
+  if [[ $CONAN_PROFILE == "iwyu" ]]; then
+    pushd ${REPO_ROOT} > /dev/null
+
+    echo -e "\n\n\nHere is the full list of all things iwyu found:"
+    cat build/iwyu_results.txt
+
+    echo -e "\n\n\nThe following include problems occur in files this PR touched:"
+    readonly REFERENCE="${KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH:-origin/master}"
+    readonly MERGE_BASE="$(git merge-base $REFERENCE HEAD)" # Merge base is the commit on master this PR was branched from.
+
+    PATTERN_FILE="$(mktemp)"
+    git diff -U0 --no-color --relative --name-only $MERGE_BASE | sed -e 's/$/:/' > ${PATTERN_FILE}
+    if grep -A1 -F "error:" build/iwyu_results.txt | grep -A1 -F -f ${PATTERN_FILE}; then
+      echo -e "\n\n"
+
+      # Replace `true` by `exit 1` here when the check is ready to fail in production!
```

Comment:
I'd prefer to start having the CI failing for touched files from now. WDYT?

Reply:
I long debated that with myself. In the end I decided against it because I expect iwyu not to be correct in many cases yet.
The mapping file for external libraries is probably not complete yet. I tried to work through all the edge cases but it's hard
because there are so many includes missing.

So I'd prefer to not fail for the next couple of days, because otherwise folks have to submit with the test failing. And if that
phase takes too long folks will stop checking the logfile because they assume iwyu failed for the usual reasons.

Reply:
Okay, I am fine with this for a couple of days :). But we really should aim to enable this. Once again, thanks a lot.